x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/jinaai/rerank/JinaAIRerankServiceSettingsTests.java

```diff
@@ -0,0 +1,99 @@
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.inference.MatchersUtils.equalToIgnoringWhitespaceInJsonString;
+
+public class JinaAIRerankServiceSettingsTests extends AbstractBWCWireSerializationTestCase<JinaAIRerankServiceSettings> {
+    public static JinaAIRerankServiceSettings createRandom() {
+        return createRandom(randomFrom(new RateLimitSettings[] { null, RateLimitSettingsTests.createRandom() }));
+    }
+
+    public static JinaAIRerankServiceSettings createRandom(@Nullable RateLimitSettings rateLimitSettings) {
+        return new JinaAIRerankServiceSettings(
+            new JinaAIServiceSettings(
+                randomFrom(new String[] { null, Strings.format("http://%s.com", randomAlphaOfLength(8)) }),
+                randomFrom(new String[] { null, randomAlphaOfLength(10) }),
+                rateLimitSettings
+            )
+        );
+    }
+
+    public void testToXContent_WritesAllValues() throws IOException {
+        var url = "http://www.abc.com";
+        var model = "model";
+
+        var serviceSettings = new JinaAIRerankServiceSettings(new JinaAIServiceSettings(url, model, null));
+
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        serviceSettings.toXContent(builder, null);
+        String xContentResult = Strings.toString(builder);
+
+        assertThat(xContentResult, equalToIgnoringWhitespaceInJsonString("""
+            {
+                "url":"http://www.abc.com",
+                "model_id":"model",
+                "rate_limit": {
+                    "requests_per_minute": 2000
+                }
+            }
+            """));
+    }
+
+    @Override
+    protected Writeable.Reader<JinaAIRerankServiceSettings> instanceReader() {
+        return JinaAIRerankServiceSettings::new;
+    }
+
+    @Override
+    protected JinaAIRerankServiceSettings createTestInstance() {
+        return createRandom();
+    }
+
+    @Override
+    protected JinaAIRerankServiceSettings mutateInstance(JinaAIRerankServiceSettings instance) throws IOException {
+        return randomValueOtherThan(instance, JinaAIRerankServiceSettingsTests::createRandom);
+    }
+
+    @Override
+    protected JinaAIRerankServiceSettings mutateInstanceForVersion(JinaAIRerankServiceSettings instance, TransportVersion version) {
+        if (version.before(TransportVersions.JINA_AI_INTEGRATION_ADDED)) {
+            // We always default to the same rate limit settings, if a node is on a version before rate limits were introduced
+            return new JinaAIRerankServiceSettings(
+                new JinaAIServiceSettings(
+                    instance.getCommonSettings().uri(),
+                    instance.modelId(),
+                    JinaAIServiceSettings.DEFAULT_RATE_LIMIT_SETTINGS
+                )
+            );
+        }
+        return instance;
```

Comment:
```suggestion
        return instance;
```

I think you can simply return `instance` here without any transport version specific mutations. As elasticsearch nodes with a version before `TransportVersions.JINA_AI_INTEGRATION_ADDED` won't have the JinaAI integration we're 100% certain that the JinaAI integration supports rate limiting.